### PR TITLE
Transactions and import diff

### LINKED ISSUE #41: Input value is not minimum needed ETH
Make input value be the minimum needed ETH instead of the sum of each transaction's value.
Should be `totalInputValues - totalOutputValues`.

https://github.com/DarkFlorist/bouquet/blob/main/app/ts/stores.ts#L28
https://github.com/DarkFlorist/bouquet/blob/main/app/ts/components/Import.tsx#L47

### LINKED ISSUE #49: Show if interceptor simulation is different than loaded tx stack

### PR DESCRIPTION
**Pull request by 0xjimmy**
* Component style consistent with new style
* Shows correct input values needed from funding transactions (Fixes #41 )
* Simplified flow to check and display ABI decoded transactions, partially fixes problems with #50, remaining issue with this is on the Configure component
* Shows warning box if Interceptor has different simulation than the current transaction list in Bouquet. Fixes #49

### CODE REVIEW

File: app/ts/components/App.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { Import } from './Import.js'
-import { Configure } from './Configure.js'
+// import { Configure } from './Configure.js'
```
Comment (KillariDev) [inline app/ts/components/App.tsx:2]: why this is commented out and not just removed?

File: app/ts/components/App.tsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ export function App() {
 						<>
 							<Import {...state} />
 							{state.bundle.value ? <Transactions {...state} /> : null}
-							<Configure {...state} />
+							{/* <Configure {...state} /> */}
```
Comment (KillariDev) [inline app/ts/components/App.tsx:31]: why not removed?

File: app/ts/library/utils.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { getAddress } from 'ethers';
+import { EthereumAddress } from '../types/ethereumTypes.js';
+import { serialize } from '../types/types.js';
```
Comment (KillariDev) [inline app/ts/library/utils.ts:3]:
```suggestion
import { getAddress } from 'ethers'
import { EthereumAddress } from '../types/ethereumTypes.js'
import { serialize } from '../types/types.js'
```

File: app/ts/library/utils.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+
+export function addressString(address: bigint): string {
+	return getAddress(serialize(EthereumAddress, address).toLowerCase())
```
Comment (KillariDev) [inline app/ts/library/utils.ts:6]:
```suggestion
	return getAddress(serialize(EthereumAddress, address))
```

serialize produces lowercased string already, and even if it didn't, getAddress can handle non lower cased one too

File: app/ts/types/bouquetTypes.ts [hunk 5]
```diff
@@ -3,7 +3,7 @@ import { EthereumAddress, EthereumInput, EthereumQuantity } from './ethereumType
 
 export type TransactionList = t.Static<typeof TransactionList>
 export const TransactionList = t.ReadonlyArray(t.Object({
-	from: EthereumAddress,
+	from: t.Union(EthereumAddress, t.Literal("FUNDING")),
```
Comment (KillariDev) [inline app/ts/types/bouquetTypes.ts:6]:
```suggestion
	from: t.Union(EthereumAddress, t.Literal('FUNDING')),
```

**Review by KillariDev — APPROVED** [review]

File: app/ts/components/App.tsx (hunk 2 above)
Comment (0xjimmy) [reply app/ts/components/App.tsx:31]: Configure has a bug and was going to fix in next PR.

File: app/ts/library/utils.ts (hunk 4 above)
Comment (0xjimmy) [reply app/ts/library/utils.ts:6]: You sure the serialize does? I know `ethers`'s getAddress would error if passed a non-checksumed string that wasn't all lowercase. Not sure if thats recently been fixed.

**Pull request merged**